drizzlepac/hlautils/sourcelist_generation.py

```diff
@@ -45,16 +52,77 @@ def create_dao_like_coordlists(fitsfile,sourcelist_filename,make_region_file=Fal
         multiplictive scale factor applied to background sigma value to compute DAOfind input parameter
         'threshold'. Default value = 2.
 
+    dao_ratio : float
+        The ratio of the minor to major axis standard deviations of the Gaussian kernel.
+
     Returns
     -------
     sources : astropy table
         Table containing x, y coordinates of identified sources
     """
+    # read in sci, wht extensions of drizzled product
     hdulist = fits.open(fitsfile)
     image = hdulist['SCI'].data
     image -= np.nanmedian(image)
+    wht_image = hdulist['WHT'].data
+
     bkg_sigma = mad_std(image, ignore_nan=True)
-    daofind = DAOStarFinder(fwhm=dao_fwhm, threshold=bkgsig_sf * bkg_sigma)
+
+    detect_sources_thresh = bkgsig_sf * bkg_sigma
+    default_fwhm = param_dict['dao']['TWEAK_FWHMPSF'] / param_dict['astrodrizzle']['SCALE']
+
+    # Estimate background for DaoStarfinder 'threshold' input.
+    bkg_estimator = MedianBackground()
+    bkg = None
+    threshold = param_dict['dao']['TWEAK_THRESHOLD']
+    exclude_percentiles = [10, 25, 50, 75]
+    for percentile in exclude_percentiles:
+        try:
+            bkg = Background2D(image, (50, 50), filter_size=(3, 3),
+                               bkg_estimator=bkg_estimator,
+                               exclude_percentile=percentile)
+        except Exception:
+            bkg = None
+            continue
+        if bkg is not None:
+            # If it succeeds, stop and use that value
+            bkg_rms = (5. * bkg.background_rms)
+            bkg_rms_mean = bkg.background.mean() + 5. * bkg_rms.std()
+            default_threshold = bkg.background + bkg_rms
+            if threshold is None:
+                threshold = default_threshold
+            elif threshold < 0:
+                threshold = -1 * threshold * default_threshold
+                log.info("{} based on {}".format(threshold.max(), default_threshold.max()))
+                bkg_rms_mean = threshold.max()
+            else:
+                bkg_rms_mean = 3. * threshold
+
+            if bkg_rms_mean < 0:
+                bkg_rms_mean = 0.
+            break
+
+    # If Background2D does not work at all, define default scalar values for
+    # the background to be used in source identification
+    if bkg is None:
+        bkg_rms_mean = max(0.01, imgarr.min())
+        bkg_rms = bkg_rms_mean * 5
+
+
+
+    # Estimate FWHM from image sources
+    kernel = astrometric_utils.build_auto_kernel(image, wht_image, threshold=bkg_rms, fwhm=default_fwhm)
+    segm = detect_sources(image, detect_sources_thresh, npixels=param_dict["sourcex"]["source_box"], filter_kernel=kernel)
+    cat = source_properties(image, segm)
+    bad_srcs = np.where(astrometric_utils.classify_sources(cat) == 0)[0] + 1
```

Comment:
This actually removes what it thinks are cosmic-rays or single-pixel events from the catalog.  I am not sure whether this is something that conforms to what the HLA-classic source identification would do.  Just wondering...

Reply:
Commenting out these two lines (117 and 118) doesn't have any impact the final coord lists produced when I ran some quick comparison tests, so I just removed those two lines in commit [94bb84d](https://github.com/spacetelescope/drizzlepac/pull/412/commits/94bb84dd98a6fface8ad10e9199512ab305e5321).